### PR TITLE
[JENKINS-31823] noProxyHost configuration is ignored

### PR DESCRIPTION
**Pull request by escoem**
Jenkins version upgraded to 1.452 and noProxyHost added to ProxyServer builder.

@reviewbybees @stephenc

### CODE REVIEW

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

**Comment by stephenc** [comment]
:bee: though better to bump to an LTS version, 1.466 or 1.480 I think are LTS baselines

**Comment by recena** [comment]
@escoem I agree with @stephenc.

File: src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
  * Utility methods for dealing with {@link com.ning.http.client.AsyncHttpClient} from a Jenkins plugin.
  *
  * @author Stephen Connolly
+ * @author Cloudbees, Inc.
```
Comment (recena) [inline src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java]: IMO, I'll remove it.

File: src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java [hunk 2]
```diff
@@ -43,6 +44,13 @@ public static ProxyServer getProxyServer() {
         if (Jenkins.getInstance() != null && Jenkins.getInstance().proxy != null) {
             final ProxyConfiguration proxy = Jenkins.getInstance().proxy;
             proxyServer = new ProxyServer(proxy.name, proxy.port, proxy.getUserName(), proxy.getPassword());
+
+            if (proxy.noProxyHost != null) {
+                for (String s : proxy.noProxyHost.split("[ \t\n,|]+")) {
+                    if (s.length()==0)  continue;
```
Comment (recena) [inline src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java]: `if (s.length() == 0)  continue;`

File: src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java (hunk 1 above)
Comment (escoem) [inline src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java]: Removed.

File: src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java (hunk 2 above)
Comment (escoem) [inline src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java]: Changed.

**Comment by escoem** [comment]
Jenkins version updated (1.466 LTS)

File: src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java [hunk 3]
```diff
@@ -23,6 +23,7 @@
  * Utility methods for dealing with {@link com.ning.http.client.AsyncHttpClient} from a Jenkins plugin.
  *
  * @author Stephen Connolly
```
Comment (stephenc) [inline src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java]: Remove all `@author` tags

**Comment by stephenc** [comment]
:bee: as it is just an :ant: that we have `@author` tags...

File: src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java (hunk 3 above)
Comment (escoem) [inline src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java]: Done.

**Comment by stephenc** [comment]
:bee:

**Comment by recena** [comment]
:bee:

File: src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java [hunk 4]
```diff
@@ -43,6 +41,14 @@ public static ProxyServer getProxyServer() {
         if (Jenkins.getInstance() != null && Jenkins.getInstance().proxy != null) {
             final ProxyConfiguration proxy = Jenkins.getInstance().proxy;
             proxyServer = new ProxyServer(proxy.name, proxy.port, proxy.getUserName(), proxy.getPassword());
+
+            if (proxy.noProxyHost != null) {
+                for (String s : proxy.noProxyHost.split("[ \t\n,|]+")) {
+                    if (s.length()>0) {
```
Comment (recena) [inline src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java]: :ant: `if (s.length() > 0) {`

**Comment by recena** [comment]
:ant:  1.466.1 (better)

File: src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java (hunk 4 above)
Comment (escoem) [inline src/main/java/jenkins/plugins/asynchttpclient/AHCUtils.java]: solved

**Comment by escoem** [comment]
The original version was 1.424 instead of 1.424.1 or 1.424.6, so I prefer 1.466 instead of 1.466.1 or 1.466.2 because there isn't any bug related to the part of the api I need. But I will agree with @stephenc as plugin maintainer.

**Comment by stephenc** [comment]
@recena I prefer to depend on the two segment versions in plugins unless there is a good reason as otherwise somebody on 1.467 might find conflicts with the changes in 1.466 -> 1.466.1 (low risk I know, but that is my reason for the preference)

**Comment by escoem** [comment]
So...

@reviewbybees done

Thanks @recena and @stephenc

**Comment by recena** [comment]
@stephenc I understand your reason. I prefer to set a LTS version (including 1.XYZ.W) because:
1. It is clear that this plugin depends on a LTS version.
2. Encourage the use of LTS versions.

**Pull request merged**